src/jit/codegenxarch.cpp

```diff
@@ -6884,6 +6993,171 @@ void        CodeGen::genCompareLong(GenTreePtr  treeNode)
+// genJTrueLong: Generate code for comparing two longs on x86 for the case where the result
+// is not manifested in a register.
+//
+// Arguments:
+//    treeNode - the compare tree
+//
+// Return Value:
+//    None.
+// Comments:
+// For long compares, we need to compare the high parts of operands first, then the low parts.
+// We only have to do the low compare if the high parts of the operands are equal.
+//
+// In the case where the result of a rel-op is not realized in a register, we generate:
+// 
+//    Opcode            x86 equivalent          Comment
+//    ------            --------------          -------
+//
+//    GT_LT; unsigned   cmp hiOp1,hiOp
+//                      jb  trueLabel
+//                      ja  labelFinal
+//                      cmp loOp1,loOp2
+//                      jb  trueLabel
+//                      labelFinal:
+//
+//    GT_LE; unsigned   cmp hiOp1,hiOp2
+//                      jb  trueLabel
+//                      ja  labelFinal
+//                      cmp loOp1,loOp2
+//                      jbe trueLabel
+//                      labelFinal:
+//    
+//    GT_GT; unsigned   cmp hiOp1,hiOp2
+//                      ja  trueLabel
+//                      jb  labelFinal
+//                      cmp loOp1,loOp2
+//                      ja  trueLabel
+//                      labelFinal:
+//
+//    GT_GE; unsigned   cmp hiOp1,hiOp2
+//                      ja  trueLabel
+//                      jb  labelFinal
+//                      cmp loOp1,loOp2
+//                      jae trueLabel
+//                      labelFinal:
+//
+//    GT_LT; signed     cmp hiOp1,hiOp
+//                      jl  trueLabel
+//                      jg  labelFinal
+//                      cmp loOp1,loOp2
+//                      jb  trueLabel
+//                      labelFinal:
+//
+//    GT_LE; signed     cmp hiOp1,hiOp2
+//                      jl  trueLabel
+//                      jg  labelFinal
+//                      cmp loOp1,loOp2
+//                      jbe trueLabel
+//                      labelFinal:
+//    
+//    GT_GT; signed     cmp hiOp1,hiOp2
+//                      jg  trueLabel
+//                      jl  labelFinal
+//                      cmp loOp1,loOp2
+//                      ja  trueLabel
+//                      labelFinal:
+//
+//    GT_GE; signed     cmp hiOp1,hiOp2
+//                      jg  trueLabel
+//                      jl  labelFinal
+//                      cmp loOp1,loOp2
+//                      jae trueLabel
+//                      labelFinal:
+//
+//    GT_EQ; signed     cmp hiOp1,hiOp2
+//                      jne labelFinal
+//                      cmp loOp1,loOp2
+//                      je  trueLabel
+//                      labelFinal:
+//
+//    GT_NE; signed     cmp hiOp1,hiOp2
+//                      jne labelTrue
+//                      cmp loOp1,loOp2
+//                      jne trueLabel
+//                      labelFinal:
+//
```

Comment:
1. GT_EQ/GT_NE handle both signed and unsigned. No need to mark them "; signed". 2. Instead of "labelFinal", use "falseLabel"? 3. there is some inconsistency between "trueLabel" and "labelTrue".